app/test_main.py

```diff
@@ -1 +1,44 @@
-# write your code here
+import pytest
+from unittest.mock import patch
+from typing import Callable
+from app.main import can_access_google_page
+
+
+@pytest.fixture
+def valid_google_url_mock() -> None:
+    with patch("app.main.valid_google_url") as mock:
+        yield mock
+
+
+@pytest.fixture
+def has_internet_connection_mock() -> None:
+    with patch("app.main.has_internet_connection") as mock:
+        yield mock
```

Comment:
there is no need to create fixtures , you can simply use  the @patch decorator. It  is more straightforward and can be easier to read